### PR TITLE
Update user login regex to match Panoptes

### PR DESCRIPTION
**Pull request by astopy**

### CODE REVIEW

**Comment by marten** [comment]
Semi-related, but I wonder if https://github.com/zooniverse/Talk-Api/blob/master/lib/username_completion.rb#L10 should get a similar treatment?

**Comment by astopy** [comment]
Hmm, yeah that should really only be defined in one place.

File: app/models/concerns/comment/mentioning.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ module Mentioning
           team
         )
       ) |
-      ( @([-\w\d]+) )     # match mentioned users
+      ( @(#{User.ALLOWED_LOGIN_CHARACTERS}+) )     # match mentioned users
```
Comment (camallen) [inline app/models/concerns/comment/mentioning.rb]: @astopy access the constant via `User::ALLOWED_LOGIN_CHARACTERS` not with the dot notation for a method.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 class User < ActiveRecord::Base
   include Moderatable
 
+  ALLOWED_LOGIN_CHARACTERS = '[\w\-\.]'
```
Comment (camallen) [inline app/models/user.rb]: can you add `'[\w\-\.]'.freeze` here to freeze the constant string?

**Comment by astopy** [comment]
@camallen OK, tests all fixed now!

**Pull request merged**